src/libs/Pusher/library/index.native.ts

```diff
@@ -2,6 +2,7 @@
  * We use the pusher-js/react-native module to support pusher on native environments.
  * @see: https://github.com/pusher/pusher-js
  */
-import Pusher from 'pusher-js/react-native';
+import PusherNative from 'pusher-js/react-native';
+import Pusher from './types';
 
-export default Pusher;
+export default PusherNative satisfies Pusher;
```

Comment:
From TS guidelines:

> Do not use [`satisfies` operator](#satisfies-operator) for platform-specific implementations, always define shared types that complies with all variants.

Reply:
Hmm , do you have an idea how to type it then? in this case platform specific files exports classes not functions , I dont see better solution for that 🤔

Reply:
How about this:
```ts
import type Pusher from './types.ts';
import PusherImplementation from 'pusher-js/with-encryption';

const PusherWeb: Pusher = PusherImplementation;

export default PusherWeb;
```